### DISCUSSION TITLE
access json datasource from transformer?

### DISCUSSION THREAD

**Discussion by dheise-rheinenergie-trading-com** [Budibase/budibase#3527] [Help]
Hi all,

I have an JSON datasource which gives me multiple rows who contains an id for a category. I also have a second JSON datasource with all my category ids and names. How can I combine both datasources to one so that I can show up the data from source 1 including the category names from source 2.

**source 1:**
`
{
  "enabled": 1,
  "timing": false,
...
  "title": "Test",
  "category": "ckvtp2tktgd",
  "target": "gk5889mdi08",
  "algo": "round_robin",
...
}
`

**source 2:**
`
{
  "title": "my title",
  "id": "ckvtp2tktgd",
...
}
`

**Transformer**
`
const rows = data.rows
for (let row of rows) {
	row['category_name'] = ... /* title from source 2 */
	
}
return rows
`

thanks,
Daniel

**Comment by mjashanks** [comment]
Hi Daniel.

I managed to do this, in the design section - although it will depend on how you want to data displayed. There's no way to get this data returned directly to a `Data Provider`, so probably won't be usable in a table.

My component structure looks like this:
![image](https://user-images.githubusercontent.com/3524181/143313461-9e01d1be-2dbe-4023-98d2-6b17a0011812.png)

"Parent Provider" would be bound to your `source 2`
"Child Provider` would be bound to you `source `

Then, inside my paragraph's text binding, I am using Javascript to lookup a value in the parent, then take the first value and return it's `Name`, for the text....

```javascript
return $("Parent Data Provider.Rows")
		.filter(parent => parent.Id === $("New Repeater.Child.ParentId"))[0].Name;
```
